infra/docker-evm-blockhain/start.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# Set log file
+LOG_FILE="debug.log"
+
+# Clear log file content if it exists
+> $LOG_FILE
+
+# Set network ID, block time, and ports
+NETWORK_ID=14578
+BLOCK_TIME=2
+
+BOOTNODE_PORT=30301
+NODE1_PORT=30310
+NODE1_HTTP_PORT=8040
+NODE1_AUTH_PORT=8551
+NODE2_PORT=30311
+NODE2_HTTP_PORT=8041
+NODE2_AUTH_PORT=8552
+
+# Create directories for the network, nodes, and bootnode
+mkdir -p bootnode
+mkdir -p node1
+mkdir -p node2
+
+# Generate genesis.json file
+cat << EOF > genesis.json
+{
+  "config": {
+    "chainId": $NETWORK_ID,
+    "homesteadBlock": 0,
+    "eip150Block": 0,
+    "eip155Block": 0,
+    "eip158Block": 0,
+    "byzantiumBlock": 0,
+    "constantinopleBlock": 0,
+    "petersburgBlock": 0,
+    "istanbulBlock": 0,
+    "berlinBlock": 0,
+    "clique": {
+      "period": $BLOCK_TIME,
+      "epoch": 30000
+    }
+  },
+  "difficulty": "1",
+  "gasLimit": "8000000",
+  "extradata": "0x0000000000000000000000000000000000000000000000000000000000000000<node1_address>0000000000000000000000000000000000000000<node2_address>0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+  "alloc": {
+    "<node1_address>": {
+      "balance": "100000000000000000000000"
+    },
+    "<node2_address>": {
+      "balance": "100000000000000000000000"
+    }
+  },
+  "coinbase": "0x0000000000000000000000000000000000000000",
+  "mixhash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+  "timestamp": "0x00"
+}
+EOF
+
+# Generate node1 account
+NODE1_PASSWORD=$(cat root-password.txt)
+echo $NODE1_PASSWORD > node1/password.txt
+geth account new --datadir node1 --password node1/password.txt >> $LOG_FILE 2>&1
+NODE1_ADDRESS=$(geth --datadir node1 account list | head -1 | awk -F'[{}]' '{print $2}')
+
+# Generate node2 account
+NODE2_PASSWORD=$(cat root-password.txt)
+echo $NODE2_PASSWORD > node2/password.txt
+geth account new --datadir node2 --password node2/password.txt >> $LOG_FILE 2>&1
+NODE2_ADDRESS=$(geth --datadir node2 account list | head -1 | awk -F'[{}]' '{print $2}')
+
+# Replace node addresses in the genesis.json file
+sed -i "s/<node1_address>/$NODE1_ADDRESS/g" genesis.json
+sed -i "s/<node2_address>/$NODE2_ADDRESS/g" genesis.json
+
+# Initialize node1 and node2 with the genesis.json file
+geth --datadir node1 init genesis.json >> $LOG_FILE 2>&1
+geth --datadir node2 init genesis.json >> $LOG_FILE 2>&1
+
+# Generate bootnode key
+bootnode --genkey bootnode/boot.key >> $LOG_FILE 2>&1
+
+# Start bootnode
+bootnode --nodekey bootnode/boot.key --verbosity 7 --addr 127.0.0.1:$BOOTNODE_PORT >> $LOG_FILE 2>&1 &
+BOOTNODE_PID=$!
+BOOTNODE_ENODE="enode://$(bootnode --nodekeyhex "$(cat bootnode/boot.key)" --writeaddress)@127.0.0.1:$BOOTNODE_PORT"
+
+# Start node1
```

Comment:
Why these are called node1 and node2? Is that the normal terminology? (not as primary-secondary, master-slave, etc?)

Reply:
Yes, in the context of a blockchain network, it is common to refer to individual participants as nodes. In this case, we have one bootnode and two additional nodes, which are referred to as node1 and node2.